google/cloud/bigquery/table.py

```diff
@@ -883,7 +883,8 @@ def upload_from_file(self,
             file_obj.seek(0, os.SEEK_SET)
 
         mode = getattr(file_obj, 'mode', None)
-        if mode is not None and mode != 'rb':
+
+        if mode is not None and mode.replace('+', '') != 'rb':
```

Comment:
I did that at first but it the tests failed in python3, which appears to normalise 'r+b' to `rb+`.